### PR TITLE
Fix #4068: isMetaKey check based on OS

### LINKED ISSUE #4068: MenuItem: link and command at the same time
PrimeFaces version: 6.3-SNAPSHOT

Right now declaring a menuitem such as:

    <p:menuitem value="Item" url="<url>" action="<action>" />

will result in only the url having effect. The menuitem will be just a link.

Instead it could both. If you simply click it, it would behave like an AJAX action, but if you middle-click or right-click and open in a new tab - then it's a link.

Why would you want that? You know how Jira has the "Create issue" button - if you simply click it, it opens the "Create issue" dialog; if you open it in a new tab, it opens the functionally identical "Create issue" page.

### PR DESCRIPTION
**Pull request by melloware**
Fix #4068: isMetaKey check based on OS

### CODE REVIEW

File: primefaces/src/main/resources/META-INF/resources/primefaces/core/core.utils.js [hunk 1]
```diff
@@ -561,6 +561,15 @@ if (!PrimeFaces.utils) {
...
             }
         },
+        
+        /**
+         * Is this CMD on MacOs or CTRL key on other OSes. 
+         * @param {JQuery.TriggeredEvent} e The key event that occurred.
+         * @return {boolean} `true` if the key is a meta key, or `false` otherwise.
+         */
+        isMetaKey: function(e) {
+            return PrimeFaces.env.browser.mac ? event.metaKey : event.ctrlKey;
```
Comment (jepsar) [inline primefaces/src/main/resources/META-INF/resources/primefaces/core/core.utils.js]:
```suggestion
            return PrimeFaces.env.browser.mac ? e.metaKey : e.ctrlKey;
```

File: primefaces/src/main/java/org/primefaces/renderkit/MenuItemAwareRenderer.java [hunk 2]
```diff
@@ -114,7 +114,7 @@ protected void encodeOnClick(FacesContext context, UIComponent source, MenuItem
 
             if (isLink) {
                 // allow CTRL+CLICK link to open new tab
-                command = "if(event.ctrlKey||event.metaKey){return true};" + command;
+                command = "if(PrimeFaces.utils.isMetaKey(event)){return true};" + command;
```
Comment (jepsar) [inline primefaces/src/main/java/org/primefaces/renderkit/MenuItemAwareRenderer.java]: It's not possible to have this as `PF.isMetaKey` (which is way shorter)?

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/renderkit/MenuItemAwareRenderer.java]: PF is only for widgets so the only shorter way would be `PrimeFaces` only but then this is a utility and belongs in the utils namespace

```js
    PF = function(widgetVar) {
    	var widgetInstance = PrimeFaces.widgets[widgetVar];

    	if (!widgetInstance) {
	        PrimeFaces.widgetNotAvailable(widgetVar);
    	}

        return widgetInstance;
    };
```

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/renderkit/MenuItemAwareRenderer.java]: OK check out my latest commit.

Comment (jepsar) [reply primefaces/src/main/java/org/primefaces/renderkit/MenuItemAwareRenderer.java]: > PF is only for widgets ...

I don't know what made me think it was an object. 😅

**Review by jepsar — APPROVED** [review]

**Pull request merged**